### ISSUE TITLE
ORLEANS0012: Report error on duplicate Id's 

### ISSUE THREAD

**Issue by ledjon-behluli** [dotnet/orleans#8663]
Create analyzer which detects duplicate IdAttribute within types.

Code fixer: No fixer (too risky)

Reason: We can never know (via static analysis) the phase on which a duplicated [Id(X)] has been applied to a member of the declaring type. This means that this type may have been serialized before by the runtime, and now a new member (having duplicate Id) is being added which if we applied a code fixer, we could override a previous member's Id to match the new structure, which could break serialization.

Key: ORLEANS0012
Category: Usage
Severity: Error

cc @ReubenBond : as per convo in #8643

**Comment by ledjon-behluli** [comment]
@ReubenBond I thought about this one, specifically in relation to the fixer (or lack of). 

Suppose at T1 we have:
```
public class C
{
[Id(0)] public string A {get;set;}
[Id(1)] public string B {get;set;}
[Id(2)] public string C {get;set;}
}
```

In point T2:
```
public class C
{
[Id(0)] public string A {get;set;}
//[Id(1)] public string B {get;set;} --> Removed
[Id(2)] public string C {get;set;}
}
```

In point T3:
```
public class C
{
[Id(0)] public string A {get;set;}
//[Id(1)] public string B {get;set;} -> Removed
[Id(2)] public string C {get;set;}
[Id(2)] public string D {get;set;} // ->Anlayzer
}
```

Still point T3:
```
public class C
{
[Id(0)] public string A {get;set;}
//[Id(1)] public string B {get;set;} -> Still removed
[Id(2)] public string C {get;set;}
[Id(1)] public string D {get;set;} // -> Fixer applied "1" as its available
}
```
To my best knowledge, the above fix would not be correct, as a previous member already had "1".

What would you say its the correct way to handle a potential fixer for this one:
1. Would it be ok for D to now have [Id(1)]?
2. If 1 is false, would it make sense to create a fixer that puts [Id(3)]?

**Comment by ledjon-behluli** [comment]
@ReubenBond can we agree upon whether its 1 or 2 in the comment above, so we can proceed further?
BTW: If no 2 is the case, then I have it ready locally and just need to push

**Comment by ReubenBond** [comment]
@ledjon-behluli IMO, it should find the max value and increment from there. This is what the code-fix to add [Id(x)] attrs does, AFAICR

**Comment by ledjon-behluli** [comment]
got it, I have a fixer locally that does that, I'll push changes later today

**Comment by ReubenBond** [comment]
Fixed by #8662
